libraries/AP_DDS/AP_DDS_Topic_Table.h

```diff
@@ -4,130 +4,120 @@
 #include "sensor_msgs/msg/BatteryState.h"
 #include "geographic_msgs/msg/GeoPoseStamped.h"
 
-#include "AP_DDS_Generic_Fn_T.h"
 #include "uxr/client/client.h"
 
 // Code generated table based on the enabled topics.
 // Mavgen is using python, loops are not readable.
 // Can use jinja to template (like Flask)
 
+enum class TopicIndex: uint8_t {
+    TIME_PUB,
+    NAV_SAT_FIX_PUB,
+    STATIC_TRANSFORMS_PUB,
+    BATTERY_STATE_PUB,
+    LOCAL_POSE_PUB,
+    LOCAL_VELOCITY_PUB,
+    GEOPOSE_PUB,
+    CLOCK_PUB,
+    JOY_SUB,
+};
+
+static inline constexpr uint8_t to_underlying(const TopicIndex index)
+{
+    static_assert(sizeof(index) == sizeof(uint8_t));
+    return static_cast<uint8_t>(index);
+}
+
```

Comment:
Correct, trying to achieve the closest looks-like-the-stil while abiding by the developer guide of no stl and no templates. 

Nope, the `case` statement doesn't have a `default`. If you add more than 255 entries to the enum, it will fail to compile. I don't predict we will ever have that many topics on ArduPilot.